decode.go

```diff
@@ -191,12 +196,21 @@ func (p *parser) node(kind Kind, defaultTag, tag, value string) *Node {
 }
 
 func (p *parser) parseChild(parent *Node) *Node {
-	child := p.parse()
+	child := p.parse(false)
 	parent.Content = append(parent.Content, child)
 	return child
 }
 
 func (p *parser) document() *Node {
+	if !p.event.implicit && len(p.event.head_comment) > 0 {
+		// This comment belongs to **before** the document event
+		n := p.node(DocumentNode, "", "", "")
+		// (1, 1) is not fully correct, but close enough
+		n.Line = 1
+		n.Column = 1
+		p.event.head_comment = nil
+		return n
+	}
 	n := p.node(DocumentNode, "", "", "")
 	p.doc = n
```

Comment:
Line 200 above is inside the if, and the following line should be there as well so that the behavior for the call-site remains the same. Positioning the if here would fix both.

Reply:
Fixed in 193064d01c66d9643f3e6b9dd4142a2a83db62b4.